[AG-4240] Feature(sidebar): allow hiding the sidebar containing the buttons and keeping the tool panel visible

sdwvit opened the pull request:
Add hideSideButtons option to side bar configuration

 - Add `hideSideButtons` boolean property to iSideBar interface
 - Update sideBarDefParser to include `hideSideButtons` in parsed config
 - Modify agSideBar logic to respect `hideSideButtons` for button visibility

Fixes AG-4240

sdwvit committed:
Introduce setSideButtonsDisplayed method with options

- Added new method to control side button visibility with show flag
- Delegates to sideBarButtons.setDisplayed and triggers state update
- Includes optional skipAriaHidden flag in options parameter

sdwvit committed:
Add hideSideButtons option to side bar configuration

 - Add `hideSideButtons` boolean property to iSideBar interface
 - Update sideBarDefParser to include `hideSideButtons` in parsed config
 - Modify agSideBar logic to respect `hideSideButtons` for button visibility
 - Remove deprecated `setSideButtonsDisplayed` method
 - Adjust conditional logic to prioritize `hideSideButtons` over default visibility

peterjrreynolds commented on packages/ag-grid-enterprise/src/sideBar/agSideBar.ts:
```diff
@@ -196,7 +196,11 @@ class AgSideBar extends Component implements ISideBar {
 
-        if (!!sideBarDef && !!sideBarDef.toolPanels) {
+        if (sideBarDef) {
+            const shouldHideSideButtons = sideBarDef.hideSideButtons ?? !this.sideBarButtons.isDisplayed();
```
Should drop `?? !this.sideBarButtons.isDisplayed();`

App could initially pass `hideSideButtons: true`, and then later update with `hideSideButtons: undefined`

sdwvit replied:
sounds good

sdwvit commented on packages/ag-grid-enterprise/src/sideBar/agSideBar.ts:
```diff
@@ -196,7 +196,11 @@ class AgSideBar extends Component implements ISideBar {
 
-        if (!!sideBarDef && !!sideBarDef.toolPanels) {
+        if (sideBarDef) {
+            const shouldHideSideButtons = sideBarDef.hideSideButtons ?? !this.sideBarButtons.isDisplayed();
+            this.sideBarButtons.setDisplayed(!shouldHideSideButtons);
```
```suggestion
            this.sideBarButtons.setDisplayed(!sideBarDef.hideSideButtons);
```

sdwvit committed:
Apply suggestion from @sdwvit

Commit:
- Updated side bar configuration to use 'hideButtons' instead of 'hideSideButtons'
- Modified SideBarDefParser to reflect renamed property in enterprise module
- Updated iSideBar interface to align with new property name 'hideButtons'
- Added warning about setSideBarVisible behavior when hideButtons is true
- Adjusted example code to use 'hideButtons: false' instead of deprecated variant
- Updated agSideBar implementation to check 'hideButtons' flag consistently

peterjrreynolds commented on documentation/ag-grid-docs/src/content/docs/side-bar/_examples/api/main.ts:
```diff
@@ -73,6 +73,7 @@ const gridOptions: GridOptions<IOlympicData> = {
         ],
         defaultToolPanel: 'filters',
         hiddenByDefault: true,
+        hideButtons: false,
```
This is the default value, so don't need to add to the example

peterjrreynolds commented on documentation/ag-grid-docs/src/content/docs/side-bar/index.mdoc:
```diff
@@ -73,7 +73,8 @@ const gridOptions = {
         position: 'left',
-        defaultToolPanel: 'filters'
+        defaultToolPanel: 'filters',
+        hideButtons: false,
```
Same as above

peterjrreynolds commented on documentation/ag-grid-docs/src/content/docs/side-bar/index.mdoc:
```diff
@@ -224,6 +225,10 @@ The list below details all the API methods relevant to the Tool Panel.
 
 {% apiDocumentation source="grid-api/api.json" section="accessories" names=["getSideBar","setSideBarVisible","isSideBarVisible","setSideBarPosition","openToolPanel","closeToolPanel","getOpenedToolPanel","isToolPanelShowing","refreshToolPanel", "getToolPanelInstance"] /%}
 
+{% warning %}
```
Change this to `note`

Commit:
pr comments
